plugin/src/main/java/org/autorefactor/refactoring/rules/AndroidViewHolderRefactoring.java

```diff
@@ -0,0 +1,312 @@
+
+                        thenBlock.statements().add(b.getAST().newExpressionStatement(itemAssignment));
+
+                        // replace previous fidnviewbyid with accesses to
+                        // viewHolderItem
+                        QualifiedName viewHolderItemFieldAccessQualifiedName = b.getAST().newQualifiedName(
+                                b.name(viewHolderItemVarName), b.simpleName(item.variable.getIdentifier()));
+                        r.replace(item.findViewByIdExpression, b.copy(qualifiedName));
+                    }
+                    // store viewHolderItem in convertView
+                    MethodInvocation setTagInvocation = b.invoke("convertView", "setTag",
+                            b.simpleName(viewHolderItemVarName));
+                    thenBlock.statements().add(b.getAST().newExpressionStatement(setTagInvocation));
+
+                    // retrieve viewHolderItem from convertView
+                    ifStatement.setElseStatement(b.block(b.getAST()
+                            .newExpressionStatement(b.assign(b.simpleName(viewHolderItemVarName),
+                                    Assignment.Operator.ASSIGN,
+                                    b.cast(b.type(viewHolderItemClassName), b.invoke("convertView", "getTag"))))));
+
+                }
+                r.remove(visitor.viewAssignmentStatement);
+                return DO_NOT_VISIT_SUBTREE;
+            }
+        }
+        return VISIT_SUBTREE;
+    }
+
+    private static boolean isInflateMethod(MethodInvocation node) {
+        return isMethod(node, "android.view.LayoutInflater", "inflate", "int", "android.view.ViewGroup")
+                || isMethod(node, "android.view.LayoutInflater", "inflate", "int", "android.view.ViewGroup", "boolean")
+                || isMethod(node, "android.view.LayoutInflater", "inflate", "org.xmlpull.v1.XmlPullParser",
+                        "android.view.ViewGroup")
+                || isMethod(node, "android.view.LayoutInflater", "inflate", "org.xmlpull.v1.XmlPullParser",
+                        "android.view.ViewGroup", "boolean");
+    }
+
+    private class GetViewVisitor extends ASTVisitor {
+        private SimpleName viewVariable;
+        private Statement viewAssignmentStatement;
+        private VariableDeclarationFragment viewVariableDeclarationFragment;
+        private Assignment viewVariableAssignment;
+        private ReturnStatement returnStatement;
+
+        @Override
+        public boolean visit(MethodInvocation node) {
+            if (isInflateMethod(node)) {
+                this.viewVariableDeclarationFragment = getAncestorOrNull(node, VariableDeclarationFragment.class);
+                if (viewVariableDeclarationFragment != null) {
+                    this.viewVariable = viewVariableDeclarationFragment.getName();
+                    this.viewAssignmentStatement = getAncestor(viewVariableDeclarationFragment,
+                            VariableDeclarationStatement.class);
```

Comment:
It could actually be a `VariableDeclarationExpression`...
What do we do about it?